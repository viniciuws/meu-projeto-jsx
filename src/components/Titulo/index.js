import React from 'react';

const Titulo = (props) => {
  const style = {
    color: props.textColor,
    width: '100%',
    textAlign: 'center'
  }
  return (
    <h1
      style={style}
    >
      {props.children}
    </h1>
  )
};

export default Titulo;