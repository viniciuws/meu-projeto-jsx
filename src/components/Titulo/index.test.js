
import React from 'react';
import testRenderer from 'react-test-renderer'
import { shallow } from 'enzyme';

import Titulo from '../Titulo'

describe('Shalow tests', () => {
  test('Testa se o componente interno é h1', () => {
    const component = shallow(<Titulo>Meu Título</Titulo>);
    const element = component.getElement();
    expect(element.type).toBe('h1');
  })
  test('Testa a cor do texto', () => {
    const component = shallow(<Titulo textColor="green">Meu Título</Titulo>);
    const styleColor = component.props().style.color;
    expect(styleColor).toBe('green');
  })
  test('Testa o texto de saída do componente', () => {
    const component = shallow(<Titulo textColor="green">Meu Título</Titulo>);
    const text = component.text();
    expect(text).toBe('Meu Título');
  })
});

describe('Snapshot tests', () => {
  test('se o componente renderiza de acordo com a snapshot', () => {
    const tree = testRenderer
      .create(<Titulo>Meu título</Titulo>)
      .toJSON();
    expect(tree).toMatchSnapshot();

  })
})