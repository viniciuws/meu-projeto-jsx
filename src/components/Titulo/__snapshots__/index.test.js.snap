// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests se o componente renderiza de acordo co ma snapshot 1`] = `
<h1
  style={
    Object {
      "color": undefined,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  Meu título
</h1>
`;

exports[`Snapshot tests se o componente renderiza de acordo com a snapshot 1`] = `
<h1
  style={
    Object {
      "color": undefined,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  Meu título
</h1>
`;
